sumo/src/raw.rs

```diff
@@ -0,0 +1,272 @@
+//! Parse SUMO networks from XML without making any simplifications or transformations. A
+//! subset of the structures and fields defined at
+//! <https://sumo.dlr.de/docs/Networks/PlainXML.html> are produced.
+
+use std::error::Error;
+
+use abstutil::Timer;
+use serde::Deserialize;
+
+use geom::{Bounds, Distance, GPSBounds, PolyLine, Polygon, Pt2D, Ring, Speed};
+
+#[derive(Deserialize)]
+pub struct Network {
+    pub location: Location,
+    #[serde(rename = "type")]
+    pub types: Vec<Type>,
+    #[serde(rename = "edge")]
+    pub edges: Vec<Edge>,
+    #[serde(rename = "junction")]
+    pub junctions: Vec<Junction>,
+    #[serde(rename = "connection")]
+    pub connections: Vec<Connection>,
+}
+
+#[derive(Deserialize)]
+pub struct Location {
+    #[serde(rename = "convBoundary", deserialize_with = "parse_bounds")]
+    pub converted_boundary: Bounds,
+    #[serde(rename = "origBoundary", deserialize_with = "parse_gps_bounds")]
+    pub orig_boundary: GPSBounds,
+}
+
+impl Network {
+    pub fn parse(path: &str, timer: &mut Timer) -> Result<Network, Box<dyn Error>> {
+        timer.start(format!("read {}", path));
+        let bytes = abstutil::slurp_file(path)?;
+        let raw_string = std::str::from_utf8(&bytes)?;
+        let network = quick_xml::de::from_str(raw_string)?;
+        timer.stop(format!("read {}", path));
+        Ok(network)
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Deserialize)]
+pub struct EdgeID(pub String);
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Deserialize)]
+pub struct NodeID(String);
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Deserialize)]
+pub struct LaneID(String);
+
+#[derive(Deserialize)]
+pub struct Type {
+    pub id: String,
+    pub priority: usize,
+    pub speed: Speed,
+    pub width: Option<Distance>,
+    #[serde(deserialize_with = "parse_list", default)]
+    pub allow: Vec<String>,
```

Comment:
Ultimately it seems like we treat this as an enum - can we parse directly to an enum here or is it not actually enummish?